pkg/sdk/alert.go

```diff
@@ -0,0 +1,421 @@
+package sdk
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+)
+
+// Compile-time proof of interface implementation.
+var _ Alerts = (*alerts)(nil)
+
+type Alerts interface {
+	// Create creates a new alert.
+	Create(ctx context.Context, id SchemaObjectIdentifier, warehouse AccountObjectIdentifier, schedule AlertSchedule, condition string, action string, opts *CreateAlertOptions) error
+	// Alter modifies an existing alert.
+	Alter(ctx context.Context, id SchemaObjectIdentifier, opts *AlterAlertOptions) error
+	// Drop removes an alert.
+	Drop(ctx context.Context, id SchemaObjectIdentifier) error
+	// Show returns a list of alerts
+	Show(ctx context.Context, opts *ShowAlertOptions) ([]*Alert, error)
+	// ShowByID returns an alert by ID
+	ShowByID(ctx context.Context, id SchemaObjectIdentifier) (*Alert, error)
+	// Describe returns the details of an alert.
+	Describe(ctx context.Context, id SchemaObjectIdentifier) (*AlertDetails, error)
+}
+
+// alerts implements Alerts
+type alerts struct {
+	client *Client
+}
+
+type CreateAlertOptions struct {
+	create      bool                   `ddl:"static" sql:"CREATE"` //lint:ignore U1000 This is used in the ddl tag
+	OrReplace   *bool                  `ddl:"keyword" sql:"OR REPLACE"`
+	alert       bool                   `ddl:"static" sql:"ALERT"` //lint:ignore U1000 This is used in the ddl tag
+	IfNotExists *bool                  `ddl:"keyword" sql:"IF NOT EXISTS"`
+	name        SchemaObjectIdentifier `ddl:"identifier"`
+
+	// required
+	warehouse AccountObjectIdentifier `ddl:"identifier,equals" sql:"WAREHOUSE"`
+	schedule  string                  `ddl:"parameter,single_quotes" sql:"SCHEDULE"`
+
+	// optional
+	Comment *string `ddl:"parameter,single_quotes" sql:"COMMENT"`
+
+	// required
+	condition AlertCondition `ddl:"condition,parentheses"   sql:"IF"`
+	action    string         `ddl:"parameter,no_equals" sql:"THEN"`
+}
+
+type AlertCondition struct {
+	Condition string `ddl:"condition,parentheses" sql:"EXISTS"`
+}
+
+func (opts *CreateAlertOptions) validate() error {
+	if !validObjectidentifier(opts.name) {
+		return errors.New("invalid object identifier")
+	}
+
+	return nil
+}
+
+func (v *alerts) Create(ctx context.Context, id SchemaObjectIdentifier, warehouse AccountObjectIdentifier, schedule AlertSchedule, condition string, action string, opts *CreateAlertOptions) error {
+	if opts == nil {
+		opts = &CreateAlertOptions{}
+	}
+	opts.name = id
+	opts.warehouse = warehouse
+	opts.schedule = schedule.String()
+	opts.name = id
+	opts.condition = AlertCondition{condition}
+	opts.action = action
+	if err := opts.validate(); err != nil {
+		return err
+	}
+	sql, err := structToSQL(opts)
+	if err != nil {
+		return err
+	}
+	_, err = v.client.exec(ctx, sql)
+	return err
+}
+
+type state string
```

Comment:
a type called "state" is too generic. Lets call it "AlertState" instead

Reply:
okay

Reply:
okay

Reply:
Ok

Reply:
fixed